Add wxStatusBar::AddFieldControl() API

AliKet opened the pull request:
Works correctly under Windows 10 and Ubuntu (wxGTK)

Commit:
Added wxStatusBar::AddFieldControl() member function

This allows user codes to simply designate where their controls should appear
in the status bar without the need to handle wxEVT_SIZE themselves. Nevertheless,
they can still handle the event if the default positionning does not meet their needs.

Commit:
Update statbar sample to show AddFieldControl() in action

vadz commented on src/common/statbar.cpp:
```diff
@@ -297,4 +299,50 @@ void wxStatusBarBase::PopStatusText(int number)
+{
+    event.Skip();
+
+    auto GetFieldNumber = [this](wxWindow* win)
```
If we stored windows instead of IDs we could simplify this by just iterating over the panes, checking if each pane has a window and repositioning the window if so.

vadz reviewed (commented):
Thanks! The API is exactly what I had in mind, so it would just need to be documented (in `interface/wx/statusbr.h`).

If the implementation can be simplified by storing `wxWindow*` (or maybe `wxWindowRef`?), I'd be for doing it, but if we have to use IDs for some reason that I'm missing, this is fine too.

Thanks again!

Commit:
Add documentation

AliKet commented on samples/statbar/statbar.cpp:
```diff
@@ -966,24 +967,7 @@ MyStatusBar::~MyStatusBar()
-
-#if wxUSE_CHECKBOX
-    wxRect rectCheck = rect;
-    rectCheck.Deflate(2);
```
It's unfortunate that there is no comment telling us why this call is needed in the first place. but testing (with and without) this `Deflate(2)` I didn't see any difference at all under Windows 10 or Linux, except under Gnome I got a warning which can be silenced by just removing this call which I did here.

vadz replied:
I'm pretty sure it was added because "it looked better" with it, but let's drop it if it's not needed any longer (in fact, I'd avoid changing the size of checkbox at all and just center it in the available space instead).

AliKet commented:
@vadz As another remark on control positioning, this code:
https://github.com/wxWidgets/wxWidgets/blob/8cd33455febb5146437912cfe2d356684824a6db/samples/statbar/statbar.cpp#L987-L992
is supposed to keep `m_statbmp` centered in the pane when the status bar is resized. But the pane is set to be fixed-width here: https://github.com/wxWidgets/wxWidgets/blob/8cd33455febb5146437912cfe2d356684824a6db/samples/statbar/statbar.cpp#L928
which simply turn off what the code above is trying to demonstrate.

So we should either use a variable-width field for the `m_statbmp` or just remove the OnSize() handler entirely and rely on the default positioning which works correctly ?

vadz commented on interface/wx/statusbr.h:
```diff
@@ -248,6 +248,22 @@ class wxStatusBar : public wxControl
+        @note You must delete the control to remove it from the status bar, as
+            simply passing @NULL will not do that.
+
+        @since 3.2.5
```
This can't be easily backported (it adds new fields), so I'd use
```suggestion
        @since 3.3.0
```

vadz reviewed (approved):
Great, thanks, I can merge it already but please let me know if you want to make more changes to the sample, e.g. make the bitmap pane growable.

AliKet committed:
Update interface/wx/statusbr.h

Apply review suggestion

Co-authored-by: VZ <vz-github@zeitlins.org>

AliKet committed:
Update statbar sample

Making the bitmap field growable for the sake of the OnSize() handle will make sense. That is, the bitmap should always be in the center of the pane.

vadz reviewed (commented):
Looks like the unused variable in the sample is breaking some builds, but otherwise it's good, thanks!

AliKet committed:
Remove unused variable after the last commit

AliKet committed:
Update interface/wx/statusbr.h

Co-authored-by: VZ <vz-github@zeitlins.org>
